### PR TITLE
EL-766: Completion time by level of help and using mode

### PR DESCRIPTION
**Pull request by patrick-laa**
[Jira ticket](https://dsdmoj.atlassian.net/browse/EL-766)

## What changed and why

- The original PR for EL-766 didn't provide separate controlled/certificated values
- Also, it provided the mean figure, which is heavily skewed by outliers, so we are now switching to the mode.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing
- Branch is generally up to date with main Github - definitely no conflicts
- No unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- PR description says *what* changed and *why*, with a link to the JIRA story.
- Diff has been checked for unexpected changes being included.
- Commit messages say why the change was made.

### CODE REVIEW

File: app/services/metrics_service.rb [hunk 1]
```diff
@@ -33,7 +33,8 @@ def metric_dataset_definition
         Geckoboard::NumberField.new(:controlled_checks_completed, name: "Controlled checks completed", optional: true),
         Geckoboard::NumberField.new(:certificated_checks_completed, name: "Certificated checks completed", optional: true),
         Geckoboard::NumberField.new(:completed_checks_per_user, name: "Completed checks per (analytics opted-in) user", optional: true),
-        Geckoboard::NumberField.new(:average_completion_time, name: "Average time taken to complete a check", optional: true),
+        Geckoboard::NumberField.new(:model_completion_time_controlled, name: "Mode of time taken to complete a controlled check", optional: true),
```
Comment (HettieS) [inline app/services/metrics_service.rb:36]:
```suggestion
        Geckoboard::NumberField.new(:mode_completion_time_controlled, name: "Mode of time taken to complete a controlled check", optional: true),
```

File: app/services/metrics_service.rb [hunk 2]
```diff
@@ -33,7 +33,8 @@ def metric_dataset_definition
...
+        Geckoboard::NumberField.new(:model_completion_time_certificated, name: "Mode of time taken to complete a certificated check", optional: true),
```
Comment (HettieS) [inline app/services/metrics_service.rb:37]:
```suggestion
        Geckoboard::NumberField.new(:mode_completion_time_certificated, name: "Mode of time taken to complete a certificated check", optional: true),
```

File: app/services/metrics_service.rb [hunk 3]
```diff
@@ -166,26 +169,37 @@ def top_validation_screens(time_period)
           .pluck(Arel.sql("page, COUNT(DISTINCT assessment_code)"))
   end
 
-  def average_completion_time(range = nil)
-    return unless relevant_events(range).where(page: "view_results").any?
+  def mode_completion_time(level_of_help, range = nil)
+    completed_checks = level_of_help == :controlled ? controlled_checks_completed(range) : certificated_checks_completed(range = nil)
```
Comment (HettieS) [inline app/services/metrics_service.rb:173]: should the range be set to nil in the certificated check method?

File: spec/services/metrics_service_spec.rb [hunk 4]
```diff
@@ -109,7 +112,8 @@
           expect(all_metric_dataset).to receive(:put).with(
             [
               {
-                average_completion_time: 1515.0,
+                mode_completion_time_controlled: 120.0,
+                mode_completion_time_certificated: 120.0,
```
Comment (HettieS) [inline spec/services/metrics_service_spec.rb:116]: can I suggest making one of these two number different or is that too fiddly with the `created_at` dates? Just to distinguish clearly that the correct value is being returned

File: spec/services/metrics_service_spec.rb [hunk 5]
```diff
@@ -133,7 +138,8 @@
                 date: Date.new(2023, 2, 1),
               },
               {
-                average_completion_time: 100.0,
+                mode_completion_time_certificated: 120.0,
+                mode_completion_time_controlled: 120.0,
```
Comment (HettieS) [inline spec/services/metrics_service_spec.rb:142]: and same again here

**Review by HettieS — COMMENTED** [review]
Looks good - a couple comments and questions

File: app/services/metrics_service.rb (hunk 3 above)
Comment (patrick-laa) [reply app/services/metrics_service.rb:173]: Nope! silly copy-paste error

File: app/services/metrics_service.rb (hunk 1 above)
Comment (patrick-laa) [reply app/services/metrics_service.rb:36]: Arg, yes, clumsy fingers

**Review by HettieS — APPROVED** [review]

**Pull request merged**